components/engine_traits/Cargo.toml

```diff
@@ -0,0 +1,31 @@
+[package]
+name = "engine_traits"
+version = "0.0.1"
+edition = "2018"
+publish = false
+
+[dependencies]
+kvproto = { git = "https://github.com/pingcap/kvproto.git" }
+protobuf = "2"
+raft = "0.6.0-alpha"
```

Comment:
I think it is not suitable for engine trait to rely on the raft.

Reply:
We need to implement `From` trait for raft::Error(see errors.rs), or has any other place to put that?

Reply:
@BusyJay Do you have some suggestions?

Reply:
yes, it is strange to depend on raft in engine...

Reply:
Hard to know a better solution when the use cases isn't here to look at.

Reply:
In cases like this `Box<Error>` seems just fine to me. It's pretty rare to ever inspect the concrete type of an error. And if the concrete type needs to be found it can be downcasted.